homeassistant/components/meteo_france/__init__.py

```diff
@@ -6,62 +6,61 @@
 from vigilancemeteo import VigilanceMeteoError, VigilanceMeteoFranceProxy
 import voluptuous as vol
 
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
 from homeassistant.const import CONF_MONITORED_CONDITIONS
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
 from homeassistant.util import Throttle
 
-from .const import CONF_CITY, DATA_METEO_FRANCE, DOMAIN, SENSOR_TYPES
+from .const import CONF_CITY, DOMAIN, SENSOR_TYPES
 
 _LOGGER = logging.getLogger(__name__)
 
 SCAN_INTERVAL = datetime.timedelta(minutes=5)
 
 
-def has_all_unique_cities(value):
-    """Validate that all cities are unique."""
-    cities = [location[CONF_CITY] for location in value]
-    vol.Schema(vol.Unique())(cities)
-    return value
-
+CITY_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_CITY): cv.string,
+        vol.Optional(CONF_MONITORED_CONDITIONS): vol.All(
+            cv.ensure_list, [vol.In(SENSOR_TYPES)]
+        ),
+    }
+)
 
 CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.All(
-            cv.ensure_list,
-            [
-                vol.Schema(
-                    {
-                        vol.Required(CONF_CITY): cv.string,
-                        vol.Optional(CONF_MONITORED_CONDITIONS): vol.All(
-                            cv.ensure_list, [vol.In(SENSOR_TYPES)]
-                        ),
-                    }
-                )
-            ],
-            has_all_unique_cities,
-        )
-    },
-    extra=vol.ALLOW_EXTRA,
+    {DOMAIN: vol.Schema(vol.All(cv.ensure_list, [CITY_SCHEMA]))}, extra=vol.ALLOW_EXTRA,
 )
 
 
-def setup(hass, config):
-    """Set up the Meteo-France component."""
-    hass.data[DATA_METEO_FRANCE] = {}
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up Meteo-France from legacy config file."""
+
+    confs = config.get(DOMAIN)
```

Comment:
Rename this variable to something more meaningful like `meteo_config` or something more commonly used like `config` or `conf`.

Reply:
I am used to write `conf` (Linky #26076, iCloud #28968, Freebox #30334), but this time I would like to emphasize that the conf is actually a list, don't know why.

Just checked and I'm the first in that case 😅

Reply:
It's just a small detail, so you can change it if you want to 😅